controller/api/public/stat_summary.go

```diff
@@ -130,50 +129,62 @@ func promLabel(key, val string) string {
 	return fmt.Sprintf("%s=\"%s\"", key, val)
 }
 
+func promNameLabel(resource *pb.Resource) string {
+	return promLabel(promResourceType(resource), resource.Name)
+}
+
+func promNamespaceLabel(resource *pb.Resource) string {
+	return promLabel(namespaceLabel, resource.Namespace)
+}
+
+func promDstLabels(resource *pb.Resource) []string {
+	return []string{
+		promLabel("dst_"+namespaceLabel, resource.Namespace),
+		promLabel("dst_"+promResourceType(resource), resource.Name),
+	}
+}
+
+func promResourceType(resource *pb.Resource) string {
+	return k8sResourceTypesToPromLabels[resource.Type]
+}
+
 func buildRequestLabels(req *pb.StatSummaryRequest) string {
 	labels := []string{}
 
 	var direction string
 	switch req.Outbound.(type) {
 	case *pb.StatSummaryRequest_OutToResource:
 		direction = "outbound"
-
-		dstLabel := fmt.Sprintf("dst_namespace=\"%s\", dst_%s=\"%s\"",
-			req.GetOutToResource().Namespace,
-			req.GetOutToResource().Type,
```

Comment:
just confirming, you alluded to fixing incorrect labels on prometheus queries. is this it? do strings returned by `req.GetOutToResource().Type` not match our `dst_*` prom labels?

Reply:
Right, without this change, we could end up with a label like `dst_deployments`, which doesn't exist.